dsgrid/config/registration_models.py

```diff
@@ -0,0 +1,129 @@
+"""Contains data models to control registration of test projects and datasets."""
+
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Any, Optional
+
+from pydantic import Field, model_validator
+from typing_extensions import Annotated
+
+from dsgrid.data_models import DSGBaseModel
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.utils.files import load_data
+
+
+class DatasetRegistrationModel(DSGBaseModel):
+    """Defines a dataset to be registered and optionally submitted to a project."""
+
+    dataset_id: Annotated[str, Field(description="Dataset ID")]
+    dataset_path: Annotated[
+        Path, Field(description="Directory containing load_data/load_data_lookup.parquet")
+    ]
+    config_file: Annotated[Path, Field(description="Path to dataset.json5")]
+    dimension_mapping_file: Annotated[
+        Optional[Path],
+        Field(
+            description="Path to file containing mappings of dataset-to-project dimensions",
+            default=None,
+        ),
+    ]
+    dimension_mapping_references_file: Annotated[
+        Optional[Path],
+        Field(
+            description="Path to file containing references to mappings of dataset-to-project dimensions",
+            default=None,
+        ),
+    ]
+    replace_dimension_names_with_ids: Annotated[
+        bool,
+        Field(
+            description="Replace the dimension entries with IDs of dimensions in the database "
+            "with matching names. Typically only useful for tests.",
+            default=False,
+        ),
+    ]
+    replace_dimension_mapping_names_with_ids: Annotated[
+        bool,
+        Field(
+            description="Replace the dimension mapping entries with IDs of dimension mappings "
+            "in the database with matching names. Typically only useful for tests.",
+            default=False,
+        ),
+    ]
```

Comment:
This is about referring to already-registered dimensions with names instead of IDs? First, the description is a bit unclear on this front. Second, are we sure this is only useful for tests? If a dimension is reused a bunch of times across datasets, could we list it using its name (and nothing else) after it gets defined once (e.g., by the project)?

Reply:
It's for obscure things like this: https://github.com/dsgrid/dsgrid-test-data/blob/main/test_efs/dsgrid_project/datasets/modeled/comstock_unpivoted/dataset.json5#L87

That is based on a very old precedent. What we are doing with real projects and datasets now is better. They always have to explicitly list the paths to dimension record files and dsgrid de-duplicates as needed. It's possible that we could remove this old stuff.